src/System.IO.Pipelines/ReadCursor.cs

```diff
@@ -157,7 +130,7 @@ private ReadCursor SeekMultiSegment(int bytes, ReadCursor end, bool checkEndReac
                 {
                     if (segmentPart.Length >= bytes)
                     {
-                        result = new ReadCursor(segmentPart.Segment, segmentPart.Start + bytes);
+                        result = new ReadCursor(segmentPart.Segment, segmentPart.Start + (int)bytes);
```

Comment:
I do not follow.
'bytes' can be as large as `segmentPart.End - segmentPart.Start`, (given the condition `if (segmentPart.Length >= bytes)`). Let's say it is.

Then:
`segmentPart.Start + (int)bytes`
= `segmentPart.Start + segmentPart.End - segmentPart.Start` => `segmentPart.End`

`segmentPart.End` can be > `segmentPart.Length`

What happens in that case?